packages/react-components/react-charts-preview/src/components/LineChart/LineChart.tsx

```diff
@@ -1,16 +1,1312 @@
+        selectedColorFillBars = [...selectedColorBarLegend, selectedColorFillBar];
+      } else {
+        selectedColorFillBars = selectedColorBarLegend
+          .slice(0, selectedColorFillBarIndex)
+          .concat(selectedColorBarLegend.slice(selectedColorFillBarIndex + 1));
+      }
+
+      const areAllColorFillBarLegendsSelected =
+        selectedColorFillBars.length === (props.colorFillBars && props.colorFillBars!.length);
+
+      if (
+        areAllColorFillBarLegendsSelected &&
+        ((props.data && props.data.lineChartData!.length === selectedLegendPoints.length) || !props.data)
+      ) {
+        // Clear all legends if all legends, including line legends, are selected
+        // Or clear all legends if all legends are selected and there is no line data
+        _clearMultipleLegendSelections();
+      } else if (!selectedColorFillBars.length && !selectedLegendPoints.length) {
+        // Clear all legends if no legends are selected, including line legends
+        _clearMultipleLegendSelections();
+      } else {
+        // set state when one or more legends are selected, including line legends
+        setSelectedColorBarLegend(selectedColorFillBars);
+        setIsSelectedLegend(true);
+      }
+
+      const selectedLegendTitlesToPass = selectedColorFillBars.map(
+        (colorFillBar: IColorFillBarsProps) => colorFillBar.legend,
+      );
+      _handleLegendClick(selectedColorFillBar, selectedLegendTitlesToPass);
+    }
+
+    function _clearMultipleLegendSelections() {
+      setSelectedColorBarLegend([]);
+      setSelectedLegendPoints([]);
+      setIsSelectedLegend(false);
+    }
+
+    /**
+     * This function checks if the given legend is highlighted or not.
+     * A legend can be highlighted in 2 ways:
+     * 1. selection: if the user clicks on it
+     * 2. hovering: if there is no selected legend and the user hovers over it*/
+
+    function _legendHighlighted(legend: string) {
+      return selectedLegend === legend || (selectedLegend === '' && activeLegend === legend);
+    }
+
+    /**
+     * This function checks if none of the legends is selected or hovered.*/
+
+    function _noLegendHighlighted() {
+      return selectedLegend === '' && activeLegend === '';
+    }
+
+    function _getColorFillBarOpacity(colorFillBar: IColorFillBarsProps) {
+      return colorFillBar.applyPattern ? 1 : 0.4;
+    }
+
+    function _getAriaLabel(lineIndex: number, pointIndex: number): string {
+      const line = _points[lineIndex];
+      const point = line.data[pointIndex];
+      const formattedDate = point.x instanceof Date ? point.x.toLocaleString() : point.x;
+      const xValue = point.xAxisCalloutData || formattedDate;
+      const legend = line.legend;
+      const yValue = point.yAxisCalloutData || point.y;
+      return point.callOutAccessibilityData?.ariaLabel || `${xValue}. ${legend}, ${yValue}.`;
+    }
+
+    function _isChartEmpty(): boolean {
+      /*return !(
```

Comment:
why comment out this code

Reply:
By mistake, I was using this to debug the code. Will change this.